core/service/storage-service.js

```diff
@@ -108,46 +108,51 @@ var StorageService = exports.StorageService = Montage.specialize({
-                        newVolblocksize.parsed = 512;
-                        dataset.properties.volblocksize = newVolblocksize;
-                        return self._dataService.getNewInstanceForType(Model.VolumeDatasetPropertyRefreservation);
-                    }).then(function(newRefreservation) {
-                        dataset.properties.refreservation = newRefreservation;
-                        return self._dataService.getNewInstanceForType(Model.VolumeDatasetPropertyReservation);
-                    }).then(function(newReservation) {
-                        dataset.properties.reservation = newReservation;
-                    });
+                    promises.push(
+                        self._populateDatasetPropertyModel(dataset, "VolumeDatasetPropertyAtime", "atime").then(function() {
+                            dataset.properties.atime.source = "INHERITED";
+                            dataset.properties.atime.parsed = "none";
+                        }),
+                        self._populateDatasetPropertyModel(dataset, "VolumeDatasetPropertyCasesensitivity", "casesensitivity").then(function(){
+                            dataset.properties.casesensitivity.source = "INHERITED";
+                            dataset.properties.casesensitivity.parsed = "none";
+                        }),
+                        self._populateDatasetPropertyModel(dataset, "VolumeDatasetPropertyCompression", "compression").then(function() {
+                            dataset.properties.compression.source = "INHERITED";
+                            dataset.properties.compression.parsed = "none";
+                        }),
+                        self._populateDatasetPropertyModel(dataset, "VolumeDatasetPropertyDedup", "dedup").then(function() {
+                            dataset.properties.dedup.source = "INHERITED";
+                            dataset.properties.dedup.parsed = "none";
+                        }),
+                        self._populateDatasetPropertyModel(dataset, "VolumeDatasetPropertyQuota", "quota"),
+                        self._populateDatasetPropertyModel(dataset, "VolumeDatasetPropertyRefquota", "refquota"),
+                        self._populateDatasetPropertyModel(dataset, "VolumeDatasetPropertyVolblocksize", "volblocksize").then(function() {
+                            dataset.properties.volblocksize.parsed = 512;
+                        }),
+                        self._populateDatasetPropertyModel(dataset, "VolumeDatasetPropertyRefreservation", "refreservation"),
+                        self._populateDatasetPropertyModel(dataset, "VolumeDatasetPropertyReservation", "reservation")
+                    );
+
+                    return Promise.all(promises);
                 });
+            } else {
+                return Promise.resolve();
             }
         }
     },
 
+    _populateDatasetProperty: {
+        value: function(dataset, modelName, propertyName) {
```

Comment:
It is more robust to pass the model object instead of just the name of the property. Moreover, it would be nice to have the name of the method to reflect the arguments, like `_populateDatasetPropertyWithModel`, and to have the arguments in the same order as the method name (dataset, property, model).